packages/cli/src/commands/set-version.ts

```diff
@@ -0,0 +1,169 @@
+import chalk from "chalk";
+import { existsSync, readFileSync, rmSync, writeFileSync } from "fs";
+import path from "path";
+import type { CommandModule } from "yargs";
+import { logError, MUDError } from "../utils/errors.js";
+
+type Options = {
+  backup?: boolean;
+  force?: boolean;
+  restore?: boolean;
+  mudVersion?: string;
+};
+
+const BACKUP_FILE = ".mudbackup";
+
+const commandModule: CommandModule<Options, Options> = {
+  command: "set-version",
+
+  describe: "Install a custom MUD version and optionally backup the previously installed version",
+
+  builder(yargs) {
+    return yargs.options({
+      backup: { type: "boolean", description: "Back up the current MUD versions to `.mudinstall`" },
+      force: {
+        type: "boolean",
+        description: "Backup fails if a .mudinstall file is found, unless --force is provided",
+      },
+      restore: { type: "boolean", description: "Restore the previous MUD versions from `.mudinstall`" },
+      mudVersion: { alias: "v", type: "string", description: "The MUD version to install" },
+    });
+  },
+
+  async handler(options) {
+    try {
+      if (!options.mudVersion && !options.restore) {
+        throw new MUDError(`Version parameter is required unless --restore is provided.`);
+      }
+
+      // Resolve the `canary` version number if needed
+      options.mudVersion =
+        options.mudVersion === "canary" ? await getCanaryVersion("@latticexyz/world") : options.mudVersion;
```

Comment:
is this just an arbitrary package for finding the current version?

Reply:
good idea, will do that!